core/src/main/java/bisq/core/api/CorePriceService.java

```diff
@@ -65,6 +86,152 @@ public double getMarketPrice(String currencyCode) throws ExecutionException, Int
+        Comparator<Offer> offerAmountComparator = Comparator.comparing(Offer::getAmount).reversed();
+
+        var buyOfferSortComparator =
+                offerPriceComparator.reversed() // Buy offers, as opposed to sell offers, are primarily sorted from high price to low.
+                        .thenComparing(offerAmountComparator);
+        var sellOfferSortComparator =
+                offerPriceComparator
+                        .thenComparing(offerAmountComparator);
+
+        List<Offer> buyOffers = openOfferManager.getOpenOffersByCurrency(Direction.BUY.name(), currencyCode).stream().map(OpenOffer::getOffer).sorted(buyOfferSortComparator).collect(Collectors.toList()); 
+        List<Offer> sellOffers = openOfferManager.getOpenOffersByCurrency(Direction.SELL.name(), currencyCode).stream().map(OpenOffer::getOffer).sorted(buyOfferSortComparator).collect(Collectors.toList());
+
+        final Optional<Offer> highestBuyPriceOffer = buyOffers.stream()
+                .filter(o -> o.getPrice() != null)
+                .max(Comparator.comparingLong(o -> o.getPrice().getValue()));
+
+        if (highestBuyPriceOffer.isPresent()) {
+            final Offer offer = highestBuyPriceOffer.get();
+        } else {
+            log.debug("highestBuyPriceOffer not present");
+        }
+
+        final Optional<Offer> highestBuyVolumeOffer = buyOffers.stream()
+                .filter(o -> o.getVolume() != null)
+                .max(Comparator.comparingLong(o -> o.getVolume().getValue()));
+
+        if (highestBuyVolumeOffer.isPresent()) {
+            final Offer offer = highestBuyVolumeOffer.get();
+        }
+
+        //DEBUG buildChartAndTableEntries called
+        double accumulatedAmount = 0;
+        
+        // Create buyer hashmap {key:price, value:count}, uses TreeMap to sort by key (asc)
+        TreeMap<Double,Double> buyTM = new TreeMap<Double,Double>();        
+        for(Offer offer: buyOffers) {            
+            // Double price = offer.getPrice().getDoubleValue() / offer.getOfferPayload().getCoinAmount();//Price per XMR coin;
+            // Long previousValue = buyTM.get(price) != null ? buyTM.get(price) : Long.valueOf(0);
+
+            Price price = offer.getPrice();
+            // buyTM.put(price, previousValue + 1);
+
+            if (price != null) {
+                double amount = (double) offer.getAmount().value / LongMath.pow(10, offer.getAmount().smallestUnitExponent());
+                accumulatedAmount += amount;
+                double priceAsDouble = (double) price.getValue() / LongMath.pow(10, price.smallestUnitExponent());
+                buyTM.put(priceAsDouble, accumulatedAmount);
+                
+                System.out.println("buy price");
+                System.out.println(priceAsDouble);
+                System.out.println("buy accumulated amount");
+                System.out.println(accumulatedAmount);
```

Comment:
please use slf4j logging if this is meant to be in the log, or remove if it's just for debugging

Reply:
done (wasn't finished debugging yet, sorry for any confusion)